src/popsift/popsift.h

```diff
@@ -74,14 +74,18 @@ class PopSift
 {
     struct Pipe
     {
-        boost::thread*                         _thread_stage1;
-        boost::thread*                         _thread_stage2;
+        std::unique_ptr<boost::thread>         _thread_stage1{nullptr};
+        std::unique_ptr<boost::thread>         _thread_stage2{nullptr};
```

Comment:
I don't think this is needed. The default constructor of unique_ptr makes sure that is "owns nothing", which should mean that it's 0.

Reply:
done d2d7618